src/components/ThemeProvider/index.js

```diff
@@ -0,0 +1,44 @@
+// @noSnapshot
+import React, { useState } from 'react'
+import PropTypes from 'prop-types'
+import { ThemeProvider as StyledThemeProvider, withTheme } from 'styled-components'
+import ResetCSS from './styles/ResetCSS'
+
+import constructTheme from './helpers/constructTheme'
+
+const ThemeProvider = ({ children, otherTheme, theme: propsTheme }) => {
+  const [theme] = useState(constructTheme(otherTheme, propsTheme))
```

Comment:
Construct a theme can take some time and doesn't need to be done at each render if the result is the same.
So, instead of useState, I think useMemo is good here :
```
const theme = useMemo(() => constructTheme(otherTheme, propsTheme), [otherTheme, propsTheme])
```

Reply:
👌 I do agree